packages/angular/projects/angular/src/components/radio-button/radio-button.strategies.ts

```diff
@@ -74,8 +74,8 @@ export interface RadioButtonStrategy {
 }
 
 export function createRadioButtonStrategy(
-  radioGroupCore$: Observable<RadioGroupCore<RadioGroupValue> | undefined> | undefined,
-  getRadioButtonValue: () => RadioGroupValue,
+  radioGroupCore$: Observable<RadioGroupCore<unknown> | undefined> | undefined,
```

Comment:
Can we make createRadioButtonStrategy generic as well?

Reply:
Yes, we can, but here are some nuances about it.

We must also complete the radio group service generic to make the radio-button strategy generic.
Because strategy uses context from the service.

```ts
// create function
export function createRadioButtonStrategy<T>(
  radioGroupCore$: Observable<RadioGroupCore<T> | undefined> | undefined, // <-- here
  getRadioButtonValue: () => unknown,
): RadioButtonStrategy {
// ...
}

// call of this function
private strategy = createRadioButtonStrategy<T>(
    this.radioGroupService?.context$, // <-- context$ must be generic with T too
    () => this.value,
  );
```

And here is the nuance comes to play :)
We know the exact T type only at runtime when the client provides value to the radio-group component.
So, in other words, we know the exact T type in the radio group at the runtime.
To provide a generic service we must know the exact T type at the compilation time.

```ts
// generic service
export const INJECTION_TOKEN = new InjectionToken('MyGenericService');

export class MyGenericService<T> {}

// providing service (register it in the DI)
@Component({
  selector: 'app-my-component',
  template: '...'
  providers: [{
    provide: INJECTION_TOKEN,
    useValue: new MyGenericService<T>, // <-- Error: T is not defined here, we must know the exact type.
  }]
})
export class MyComponent {}
```

Some workarounds exist to solve this issue but in my opinion, the game's not worth playing.
Because we don't work with type in strategy/service and the unknown type is okay here :)